Frontend Overhaul

SamEThibault opened the pull request:
I created a nav template served by the home controller that gets generated by a JS function for each page. This removed several lines of duplicate code and will allow us to easily edit the navigation bar.

I added some security exceptions to allow the stylesheets to load. There's still loads of work to be done. In terms of styling, I got rid of our previous color scheme and everything is left as a black and white template, we can test out some pallets and go from there.

The stylesheets still contain rules that should be removed due to the new frontend structure: nav bar styling on each page etc...

I noticed that though the discussion board page works well, the API returns some weird URIs when posting. From the API, it seems like it should return the original template, but it returns a getPosts endpoint or something. This could become an issue once we start styling things more.

The padding of some elements had been affected by the re-formatting of the css files, but I plan on writing a function to place the data in some Bootstrap table so it's left as-is for now.

I also noticed that the discussion board template is not in the same templates folder as the other files. Not sure where it's generated from.

One last thing, we should determine a standard for HTML document structures. Some documents had no HTML tag, and some had trailing tags. The headers should also be ordered for consistency and a proper title should be given to each page.

What needs fixing:
- Find a better way to output car data
- Add a Carousel to the explore page
- Find color pallet and roll with it for all pages (with the new structure, we can make a main.css file, import it into all the html files, and add all of the common styling rules that we want to be consistend throughout the app)
- Ensure endpoints are configured properly
- Enforce consistent HTML formatting

I'd love for you to check the code out, tweak some stuff if need-be, and I'll keep working on the above to-do list in the next few days!

Cheers

SamEThibault committed:
separate css from html, adjusted security to allow stylesheet loading

MattConstant commented:
All seems good. 

I would like to make some comments on your post to clarify some issues had with the site.

To begin, I was able to locate the issue with on the discussion board. This bug was fixed and will be released in the next pull request or whenever possible.

Secondly, the discussion template has been named "community.html". It might have been overlooked as the name does not exactly correspond to its use. Keep an eye out for it next time you are working on the site and follow up if you still don't see it.

To conclude, everything else seems to be in order I agree that we should have a discussion about using a better HTML structure. 

Nice job!

SamEThibault committed:
re-style discussion board

SamEThibault committed:
complete engine.html re-format

SamEThibault committed:
complete body.html re-format

SamEThibault committed:
complete fuel.html re-format

SamEThibault committed:
re-style community.html

SamEThibault committed:
re-style search.html

SamEThibault committed:
add carousel to explore page

SamEThibault committed:
fixed login redirection

SamEThibault committed:
fixed login signup visibility and redirects

SamEThibault committed:
rm unused logout checks on frontend, add /login parameter

SamEThibault commented:
I fixed the visibility of the login/signup/logout buttons depending on if the user is authenticated. The login prompt for the discussion board and favorites page makes sense since we shouldn't allow the user to access these pages unless they are logged in. They are, however, allowed to view everything else. 

This had something to do with the SecurityConfig file: .anyRequest().authenticated() needed to be commented out for the authentication feature to work as intended.

I noticed that the signup functionality doesn't work, I'll look into that next. 

The logout page now properly redirects to the login screen: there was an error in the SecurityConfig file where it would serve /?logout , which is not a valid endpoint, to the user. It now serves /login. I also added a url parameter so that the proper "you have been logged out" message appears accordingly.

SamEThibault committed:
cleaned up nav btns

MattConstant committed:
Fixed Login Issue

There was an issue with the login page where we were not able to verify existing users. This has been fixed
